### PR TITLE
Text fixes and some shape/free func additions

### LINKED ISSUE #108: Text placement on a 2D curve
Support text on a 2D curve

Continuation of #48

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
- [x] single letter fix
- [x] 2D text on path
- [x] text on path
- [x] text on path/face
- [x] bin brep import/export
- [x] solid constructors tweaks
- [x] symmetric offset
- [x] "infinite" plane
- [x] triangulation edge visualization

fixes #108, #109, #1581

afbeelding
afbeelding

### CODE REVIEW

File: cadquery/occ_impl/shapes.py [hunk 1]
```diff
@@ -520,6 +522,26 @@ def importBrep(cls, f: Union[str, BytesIO]) -> "Shape":
 
         return cls.cast(s)
 
+    def exportBin(self, f: Union[str, BytesIO]) -> bool:
+        """
+        Export this shape to a binary BREP file
```
Comment (jmwright) [inline cadquery/occ_impl/shapes.py:527]:
```suggestion
        Export this shape to a binary BREP file.
```

File: cadquery/occ_impl/shapes.py [hunk 2]
```diff
@@ -520,6 +522,26 @@ def importBrep(cls, f: Union[str, BytesIO]) -> "Shape":
...
+        """
+
+        rv = BinTools.Write_s(self.wrapped, f)
+
+        return True if rv is None else rv
+
+    @classmethod
+    def importBin(cls, f: Union[str, BytesIO]) -> "Shape":
+        """
+        Import shape from a binary BREP file
```
Comment (jmwright) [inline cadquery/occ_impl/shapes.py:537]:
```suggestion
        Import shape from a binary BREP file.
```

File: cadquery/vis.py [hunk 3]
```diff
@@ -145,10 +152,16 @@ def show(*objs: Showable, scale: float = 0.2, alpha: float = 1, **kwrags: Any):
     axs = _to_vtk_axs(locs, scale=scale)
 
     # create a VTK window
-    win = _vtkRenderWindow(assy)
+    win = _vtkRenderWindow(assy, tolerance=tolerance)
 
     win.SetWindowName("CQ viewer")
 
+    # get renderer and actor`
```
Comment (jmwright) [inline cadquery/vis.py:159]:
```suggestion
    # get renderer and actor
```

**Review by jmwright — APPROVED** [review]
Thanks @adam-urbanczyk !

**Comment by lorenzncode** [comment]
Not critical, but I noticed an issue with the git log with duplicated messages.  It seems that since the same branch is used for the last few PR, the messages are repeated.

For example compare the first ~30 lines of the message of these two commits:

```
$ git log --pretty=format:'%h %ad%n%s%n%b' -n 1 a175c | head -n 30 | grep -E "\S"
a175cb8 Mon Nov 11 23:21:49 2024 +0100
More shape/free func improvements (#1692)
* Initial commit of better paramAt, solid and a new function check
* mypy fix
* isSolid fix
* Implement outerShell, innerShells
* Imports cleanup
* Typo fix
* Make paramAt fully compatibile with Wires
* Typo fix
* Add check test
* Fix orientation handling
* Add test
* paramAt test
* More tests
* Add orientation fix and optimize check

$ git log --pretty=format:'%h %ad%n%s%n%b' -n 1 5720c | head -n 30 | grep -E "\S"
5720c7d Tue Oct 8 18:08:58 2024 +0200
cq.Shape and some free-func improvements (#1672)
* Initial commit of better paramAt, solid and a new function check
* mypy fix
* isSolid fix
* Implement outerShell, innerShells
* Imports cleanup
* Typo fix
* Make paramAt fully compatibile with Wires
* Typo fix
* Add check test
* Fix orientation handling
* Add test
* paramAt test
* More tests
* Add orientation fix and optimize check
```

Maybe the commit message can be manually edited when this is checked-in.

Looks interesting, I'll do some testing of the changes.

**Comment by lorenzncode** [comment]
Is it possible to create a solid from a mix of shells and faces?
```py
w1 = rect(10, 10)
sh1 = extrude(w1, (0, 0, 5))  # box sides (shell)

e1 = segment((5, -5), (5, 5))
f1 = extrude(e1, (-10, 0, 0))  # bottom face
f2 = extrude(e1.move((0, 0, 5)), (-10, 0, 0))  # top face


# create solid box from shell and faces?
# box1 = solid(...)
```

File: doc/free-func.rst [hunk 4]
```diff
@@ -237,3 +237,41 @@ Placement and creation of arrays is possible using :meth:`~cadquery.Shape.move`
     c = cylinder(1,2).move(rx=15).moved(*locs)
 
     result = compound(s, c.moved(2))
+
+Text
+----
+
+The free functon API has extensive text creation capabilities including text on planar curves and text on surfaces.
```
Comment (lorenzncode) [inline doc/free-func.rst:244]: typo: functon -> function

File: doc/free-func.rst [hunk 5]
```diff
@@ -237,3 +237,41 @@ Placement and creation of arrays is possible using :meth:`~cadquery.Shape.move`
...
+    # planar
+    r1 = text(TXT, 1, spine, planar=True).moved(z=-S)
+
+    # normal
+    r2 = text(TXT, 1, spine)
+
+    # projected
+    r3 = text(TXT, 1, spine, cf).moved(z=S)
+
+    # projected and thickend
```
Comment (lorenzncode) [inline doc/free-func.rst:274]: typo: thickend -> thickened

**Review by lorenzncode — APPROVED** [review]
I like the new features!  Thanks @adam-urbanczyk !

**Pull request merged**